plugins/workflow/notification/notification.php

```diff
@@ -0,0 +1,262 @@
+	 *
+	 * @var   CMSApplicationInterface
+	 * @since __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * Database object.
+	 *
+	 * @var    JDatabaseDriver
+	 * @since  3.9.0
+	 */
+	protected $db;
+
+	/**
+	 * The form event.
+	 *
+	 * @param   Form      $form  The form
+	 * @param   stdClass  $data  The data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onContentPrepareForm(Form $form, $data)
+	{
+		$context = $form->getName();
+
+		// Extend the transition form
+		if ($context !== 'com_workflow.transition')
+		{
+			return;
+		}
+
+		return $this->enhanceTransitionForm($form, $data);
+	}
+
+	/**
+	 * Add different parameter options to the transition view, we need when executing the transition
+	 *
+	 * @param   Form      $form  The form
+	 * @param   stdClass  $data  The data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function enhanceTransitionForm(Form $form, $data)
+	{
+		$model = $this->app->bootComponent('com_workflow')
+			->getMVCFactory()->createModel('Workflow', 'Administrator', ['ignore_request' => true]);
+
+		$workflow_id = !empty($data->workflow_id) ? (int) $data->workflow_id : (int) $form->getValue('workflow_id');
+
+		if (empty($workflow_id))
+		{
+			$workflow_id = $this->app->input->getInt('workflow_id');
+		}
+
+		$workflow = $model->getItem($workflow_id);
+
+		Form::addFormPath(__DIR__ . '/forms');
+
+		$form->loadFile('workflow_notification');
+
+		return true;
+	}
+
+	/**
+	 * Send a Notification to defined users a transion is performed
+	 *
+	 * @param   string   $context  The context for the content passed to the plugin.
+	 * @param   array    $pks      A list of primary key ids of the content that has changed stage.
+	 * @param   object  $data    Object containing data about the transition
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onWorkflowAfterTransition($context, $pks, $data)
+	{
+		// Send mail active
+		if (isset($data->options['send_mail']) && !$data->options['send_mail'])
+		{
+			return true;
+		}
+
+		// ID of the items whose state has changed. 
+		$pks = ArrayHelper::toInteger($pks);
+
+		if (empty($pks))
+		{
+			return true;
+		}
+
+		// Get UserIds of Receivers
+		$userIds = $this->getUsersFromGroup($data);
+
+		// Email to author 
+		if (!empty($data->options['email_author']))
+		{
+			$author = $this->app->getIdentity($item->created_by);
+
+			if(!empty($author) && !$author->block)
+			{
+				if (!in_array($author->id, $userIds))
+				{
+					$userIds[] = $author->id;
+				}
+			}
+		}
+
+		// If there are no receivers, stop here
+		if (empty($userIds))
+		{
+			return true;
+		}
+
+		// Prepare Language for messages
+		$default_language = ComponentHelper::getParams('com_languages')->get('administrator');
+		$debug = $this->app->get('debug_lang');
+
+		// Get the Model via $context
+		$parts = explode('.', $context);
+
+		$component = $parts[0];
+		$modelName = $parts[1];
```

Comment:
It's not guaranteed, that ```$parts[1]``` exists, needs a check before.
There is not always the correct model name in the context, use: https://github.com/bembelimen/joomla-cms/blob/workflow-v3/plugins/workflow/publishing/publishing.php#L331

Reply:
ty